### DISCUSSION TITLE
all_unread_notifications" is not a table or materialized view

### DISCUSSION THREAD

**Discussion by boilermanc** [supabase/supabase#20397] [Questions]
Trying to update a view with a trigger. I have created the function and then the trigger. the trigger listens to changes in a table and then runs the function to update the view but i get this error: all_unread_notifications" is not a table or materialized view:

BEGIN
  REFRESH MATERIALIZED VIEW all_unread_notifications;
  RETURN NULL;
END;

all_unread_notifications is the view. 

thanks!

**Comment by GaryAustin1** [comment]
Do you have a materialized view?  https://www.postgresql.org/docs/current/rules-materializedviews.html
Or a "normal" view?  https://www.postgresql.org/docs/current/tutorial-views.html

**Reply by GaryAustin1** [reply]
Regular views don't need updating.  They are just a query that presents a different "view" of the table.  This is the most common view.

Materialized views actually have to store data.  They are likely faster to read depending on complexity of query to build the view, but go stale without being updated like you were doing.

**Comment by boilermanc** [comment]
Ahh no just a regular view I guess. So I create a materialized view or is there a statement to update my regular view. Hope I got that right.

⁣Get BlueMail for Android ​

**Comment by boilermanc** [comment]
Great thank you. I'll try a materialized view and see if that does the trick.

⁣Get BlueMail for Android ​
